recipe/meta.yaml

```diff
@@ -1,67 +1,60 @@
-{% set version = "3.34.2" %}
-{% set min_python = "python >=3.8" %}
+{% set version = "3.35.0" %}
+
+# handle undefined PYTHON in `noarch: generic` outputs
+{% if PYTHON is not defined %}{% set PYTHON = "$PYTHON" %}{% endif %}
 
 package:
   name: openmdao
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/o/openmdao/openmdao-{{ version }}.tar.gz
-  sha256: 3f772f74c99fc2305a872a4761589d4130396ea98a2210d9d1367fcf7fca36ab
+  sha256: cb61513f387bb14ee6ba04f641a3dc614bc4df734251c6f738712599a159b6b0
 
 build:
   noarch: python
   number: 0
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  entry_points:
+    - openmdao = openmdao.utils.om:openmdao_cmd
+    - reset_notebook = openmdao.devtools.docutils:reset_notebook_cmd
+    - run_om_test = openmdao.devtools.run_test:run_test
+    - webview = openmdao.utils.webview:webview_argv
+    - wingproj = openmdao.devtools.wingproj:run_wing
 
 requirements:
   host:
-    - {{ min_python }}
+    - hatchling
+    - pip
+    - python >=3.9
   run:
-    - {{ min_python }}
+    - networkx >=2.0
+    - numpy
+    - packaging
+    - python >=3.9
+    - requests
+    - scipy
+
+test:
+  requires:
+    - pip
+  imports:
+    - openmdao
+  commands:
+    - pip check
+    - openmdao --help
+    - webview --help
+    - wingproj --help
 
 outputs:
   - name: openmdao
-    build:
-      noarch: python
-      script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
-      entry_points:
-        - openmdao = openmdao.utils.om:openmdao_cmd
-        - reset_notebook = openmdao.devtools.docutils:reset_notebook_cmd
-        - run_om_test = openmdao.devtools.run_test:run_test
-        - webview = openmdao.utils.webview:webview_argv
-        - wingproj = openmdao.devtools.wingproj:run_wing
-    requirements:
-      host:
-        - {{ min_python }}
-        - hatchling
-        - pip
-      run:
-        - {{ min_python }}
-        - networkx >=2.0
-        - numpy <2
-        - packaging
-        - requests
-        - scipy
-    test:
-      requires:
-        - pip
-      imports:
-        - openmdao
-      commands:
-        - pip check
-        - openmdao --help
-        - webview --help
-        - wingproj --help
 
   - name: openmdao-with-doe
     build:
-      noarch: python
+      noarch: generic
     requirements:
-      host:
-        - {{ min_python }}
       run:
-        - {{ min_python }}
-        - {{ pin_subpackage("openmdao", exact=True) }}
+        - {{ pin_subpackage("openmdao", max_pin="x.x.x") }}
```

Comment:
recently i was given the guidance that we should be using `noarch: generic` wherever possible, as it skips a bunch of solves.

however, going from `noarch: python` to `noarch: generic` changes the build hash (the garbage right before `_0.conda` in the filename). this is particularly sensitive for really big builds with variants (think `cuda`, `numpy`/`cython` epochs). using `max_pin` means we accept builds that don't have a matching hash. it _can_ introduce issues if we ship a `_1` and need to patch stuff, but often will end up doing that _anyway_ on a `repodata-patches` PR.

Reply:
(it _also_ means we only use `python >=3.9` in _exactly two places_, and they fit on the same screen, so no need for the icky python variable, which the bot always seems to screw up.